toponetx/classes/colored_hypergraph.py

```diff
@@ -1,7 +1,7 @@
 """Creation and manipulation of a Colored Hypergraph."""
 
 from collections.abc import Collection, Hashable, Iterable
-from typing import Literal
+from typing import Literal, Optional
```

Comment:
We are on Python 3.10+, `Optional` is depreciated. Use union type expressions, e.g. `str | None` instead of `Optional[str]`.